plumbing/transport/receive_pack.go

```diff
@@ -134,6 +139,16 @@ func ReceivePack(
 		return err
 	}
 
+	if useSideband {
+		if err := pktline.WriteFlush(w); err != nil {
+			return fmt.Errorf("flushing sideband: %w", err)
+		}
+	}
+
+	if err := w.Close(); err != nil {
+		return fmt.Errorf("closing writer: %w", err)
+	}
```

Comment:
This may need to be called on a defer instead, to avoid it not being called at all - as per the other comments.